wcfsetup/install/files/lib/data/I18nDatabaseObjectList.class.php

```diff
@@ -0,0 +1,72 @@
+<?php
+namespace wcf\data;
+use wcf\system\exception\SystemException;
+use wcf\system\WCF;
+use wcf\util\StringUtil;
+
+/**
+ * Abstract class for a list of database objects with better sorting of i18n-based columns.
+ *
+ * @author	Florian Gail
+ * @copyright	2001-2017 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\Data
+ * @since	3.1
+ */
+abstract class I18nDatabaseObjectList extends DatabaseObjectList {
+	/**
+	 * Array of column names, that are eventually filled with language items.
+	 * Those additional joins are going to slow down the system; you should use this
+	 * class only when it's really needed.
+	 * The key represents the original field name.
+	 * The value represents the new field name containing the localized field-value.
+	 *
+	 * @example   [ 'title' => 'titleSortField' ]
+	 *
+	 * @var string[]
+	 */
+	public $i18nFields = [];
+	
+	/**
+	 * ID of the language that should be used
+	 * the value defaults to the current user's language
+	 *
+	 * @var integer
+	 */
+	public $languageID;
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function __construct() {
+		parent::__construct();
+		
+		if ($this->languageID === null) {
+			$this->languageID = WCF::getUser()->languageID;
+		}
+		
+		if (!empty($this->i18nFields)) {
+			if (count($this->i18nFields) != count(array_flip($this->i18nFields))) {
+				throw new SystemException("'".$this->className."::\$i18nFields' should unique values.");
```

Comment:
You already 👍 the comment by @Gravatronics, but it was marked outdated for some reason: I agree with his review, but suggest the more specific `DomainException` (which is a subclass of `LogicException`), instead.

Reply:
I read the commit after pushing my commits and wanted to wait for your feedback before I mess up everything with commits. :)

Reply:
Yes, that's what I guessed. But GitHub should not have collapsed his comment as outdated, because it is newer than all the commits *and* refers to a line existing in the current diff.